Sources/PriorityQueueModule/PriorityQueue.swift

```diff
@@ -0,0 +1,537 @@
+  internal var storage: [Element]
+
+  /// A Boolean value indicating whether or not the queue is empty.
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public var isEmpty: Bool {
+    storage.isEmpty
+  }
+
+  /// The number of elements in the queue.
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public var count: Int {
+    storage.count
+  }
+
+  /// A read-only view into the underlying heap.
+  ///
+  /// In the current implementation, the elements aren't _arbitrarily_ ordered,
+  /// as a min-max heap is used for storage. However, no guarantees are given as
+  /// to the ordering of the elements.
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public var unordered: [Element] {
+    storage
+  }
+
+  /// Creates an empty queue.
+  @inlinable
+  public init() {
+    storage = []
+  }
+
+  /// Inserts the given element into the queue.
+  ///
+  /// - Complexity: O(log `count`) / 2
+  @inlinable
+  public mutating func insert(_ element: Element) {
+    storage.append(element)
+    _bubbleUp(startingAt: storage.endIndex - 1)
+    _checkInvariants()
+  }
+
+  /// Returns the element with the lowest priority, if available.
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public func min() -> Element? {
+    storage.first
+  }
+
+  /// Returns the element with the highest priority, if available.
+  ///
+  /// - Complexity: O(1)
+  @inlinable
+  public func max() -> Element? {
+    switch storage.count {
+    case 0, 1, 2:
+      // If count is 0, `last` will return `nil`
+      // If count is 1, the last (and only) item is the max
+      // If count is 2, the last item is the max (as it's the only item in the
+      // first max level)
+      return storage.last
+    default:
+      // We have at least 3 items -- return the larger of the two in the first
+      // max level
+      return Swift.max(storage[1], storage[2])
+    }
+  }
+
+  /// Removes and returns the element with the lowest priority, if available.
+  ///
+  /// - Complexity: O(log `count`)
+  @inlinable
+  public mutating func popMin() -> Element? {
+    defer { _checkInvariants() }
+    return _remove(at: 0)
+  }
+
+  /// Removes and returns the element with the highest priority, if available.
+  ///
+  /// - Complexity: O(log `count`)
+  @inlinable
+  public mutating func popMax() -> Element? {
+    defer { _checkInvariants() }
+    switch storage.count {
+    case 0, 1, 2:
+      // If count is 0, `popLast` will return `nil`
+      // If count is 1, the last (and only) item is the max
+      // If count is 2, the last item is the max (as it's the only item in the
+      // first max level)
+      return storage.popLast()
+    default:
+      // The max item is the larger of the two items in the first max level
+      let maxIdx = storage[2] > storage[1] ? 2 : 1
+      return _remove(at: maxIdx)
+    }
+  }
+
+  /// Removes and returns the element with the lowest priority.
+  ///
+  /// The queue *must not* be empty.
+  ///
+  /// - Complexity: O(log `count`)
```

Comment:
Should this complexity be O(log `count`) **/ 4**, since `insert`'s complexity includes the "/ 4" part?

Reply:
We should probably stick with https://github.com/apple/swift-collections/pull/51#issuecomment-854003019 and not include any constants here and in `insert`'s docs.